app/modules/elocaps/cruds_elocaps.py

```diff
@@ -0,0 +1,271 @@
+            .selectinload(models_elocaps.GamePlayer.player)
+            .selectinload(models_elocaps.Player.user),
+        ),
+    )
+    return result.scalars().all()
+
+
+async def get_player_info(
+    db: AsyncSession,
+    user_id: str,
+) -> Sequence[models_elocaps.Player]:
+    result = await db.execute(
+        select(models_elocaps.Player)
+        .where(models_elocaps.Player.user_id == user_id)
+        .options(selectinload(models_elocaps.Player.user)),
+    )
+    return result.scalars().all()
+
+
+async def get_player_games(db: AsyncSession, user_id: str) -> list[models_elocaps.Game]:
+    result = await db.execute(
+        select(models_elocaps.GamePlayer)
+        .join(models_elocaps.Player)
+        .where(models_elocaps.Player.user_id == user_id)
+        .options(
+            selectinload(models_elocaps.GamePlayer.game)
+            .selectinload(models_elocaps.Game.game_players)
+            .selectinload(models_elocaps.GamePlayer.player)
+            .selectinload(models_elocaps.Player.user),
+        ),
+    )
+    return [i.game for i in result.scalars().all()]
+
+
+async def get_game_details(
+    db: AsyncSession,
+    game_id: str,
+) -> models_elocaps.Game | None:
+    result = await db.execute(
+        select(models_elocaps.Game)
+        .where(models_elocaps.Game.id == game_id)
+        .options(
+            selectinload(models_elocaps.Game.game_players)
+            .selectinload(models_elocaps.GamePlayer.player)
+            .selectinload(models_elocaps.Player.user),
+        ),
+    )
+    return result.scalars().first()
+
+
+async def create_game(db: AsyncSession, game: models_elocaps.Game) -> None:
+    db.add(game)
+    await db.commit()
+
+
+async def insert_player_into_game(
+    db: AsyncSession,
+    game: models_elocaps.Game,
+    player_info: schemas_elocaps.GamePlayerBase,
+) -> None:
+    try:
+        player = (
+            (
+                await db.execute(
+                    select(models_elocaps.Player).where(
+                        (models_elocaps.Player.user_id == player_info.user_id)
+                        & (models_elocaps.Player.mode == game.mode),
+                    ),
+                )
+            )
+            .scalars()
+            .first()
+        )
+        if not player:
+            player = models_elocaps.Player(user_id=player_info.user_id, mode=game.mode)
+            db.add(player)
+            await db.commit()
+        game_player = models_elocaps.GamePlayer(
+            game_id=game.id,
+            player_id=player.id,
+            team=player_info.team,
+            score=player_info.score,
+        )
+        db.add(game_player)
+        await db.commit()
+    except IntegrityError as error:
+        await db.rollback()
+        raise ValueError(error)
+
+
+async def set_player_elo_gain(
+    db: AsyncSession,
+    game_player: models_elocaps.GamePlayer,
+    gain: int,
+) -> None:
+    game_player.elo_gain = gain
+    await db.commit()
```

Comment:
If this syntax is perfectly fine, it is not used in other modules which prefer `.update`. I'm not sure what we want to do for this.